docs/design/ovs-pipeline.md

```diff
@@ -1,1197 +1,1852 @@
-connection yet; instead we store the port in NXM_NX_REG1 - similarly to how we
-process non-Service traffic in [L2ForwardingCalcTable] - and forward it to
-[ConntrackCommitTable]. By committing the connection we ensure that reply
-traffic (traffic from the Service backend which has already gone through
-kube-proxy for source IP rewrite) will not be dropped because of Network
-Policies.
-
-The table-miss flow entry (flow 2) for this table forwards all non-Service
-traffic to the next table, [AntreaPolicyEgressRuleTable].
-
-[ClassifierTable]: #classifiertable-0
-[SpoofGuardTable]: #spoofguardtable-10
-[ARPResponderTable]: #arprespondertable-20
-[ServiceHairpinTable]: #servicehairpintable-23
-[ConntrackTable]: #conntracktable-30
-[ConntrackStateTable]: #conntrackstatetable-31
-[DNATTable]: #dnattable-40
-[SessionAffinityTable]: #sessionaffinitytable-40
-[ServiceLBTable]: #servicelbtable-41
-[EndpointDNATTable]: #endpointdnattable-42
-[AntreaPolicyEgressRuleTable]: #antreapolicyegressruletable-45
-[EgressRuleTable]: #egressruletable-50
-[EgressDefaultTable]: #egressdefaulttable-60
-[L3ForwardingTable]: #l3forwardingtable-70
-[SNATTable]: #snattable-71
-[L3DecTTLTable]: #l3decttltable-72
-[L2ForwardingCalcTable]: #l2forwardingcalctable-80
-[AntreaPolicyIngressRuleTable]: #antreapolicyingressruletable-85
-[IngressRuleTable]: #ingressruletable-90
-[IngressDefaultTable]: #ingressdefaulttable-100
-[ConntrackCommitTable]: #conntrackcommittable-105
-[HairpinSNATTable]: #hairpinsnattable-108
-[L2ForwardingOutTable]: #l2forwardingouttable-110
+Flow 1 is for case 1. It matches packets with `L7NPRedirectCTMark` and `OutputToOFPortRegMark`, and then outputs them to
+the port `antrea-l7-tap0` specifically created for connecting to an application-aware engine. Notably, these packets are pushed
+with an 802.1Q header and loaded with the VLAN ID value persisted in `L7NPRuleVlanIDCTLabel` before being output, due to
+the implementation of `L7NetworkPolicy`.
+
+Flow 2 is for case 2. It matches packets with `TrafficControlMirrorRegMark` and `OutputToOFPortRegMark`, and then
+outputs them to the port specified in `TargetOFPortField` and the port specified in `TrafficControlTargetOFPortField`.
+
+Flow 3 is for case 3. It matches packets with `TrafficControlRedirectRegMark` and `OutputToOFPortRegMark`, and then
+outputs them to the port specified in `TrafficControlTargetOFPortField`.
+
+Flow 4 is for case 4. It matches packets from hairpin connections by matching `HairpinCTMark` and outputs them back to the
+port where they were received.
+
+Flow 5 is for case 5. It matches packets by matching `OutputToOFPortRegMark` and outputs them to the OVS port specified by
+the value stored in `TargetOFPortField`.
+
+Flows 6-7 are for case 6. They match packets by matching `OutputToControllerRegMark` and the value stored in
+`PacketInOperationField`, then output them to the OpenFlow controller (Antrea Agent) with corresponding user data.
```

Comment:
Add the following at the end of the paragraph:

> In practice, you will see additional flows similar to these ones to accommodate different scenarios (different `PacketInOperationField` values). Note that packets sent to controller are metered to avoid overrunning the Antrea Agent and using too many resources.